workflow/executor/k8sapi/k8sapi.go

```diff
@@ -56,13 +57,28 @@ func (k *K8sAPIExecutor) GetExitCode(ctx context.Context, containerID string) (s
 	return "", nil
 }
 
-func (k *K8sAPIExecutor) WaitInit() error {
-	return nil
+func (k *K8sAPIExecutor) WaitInit(ctx context.Context) (containerID string, err error) {
+	return containerID, k.client.until(ctx, func(pod *corev1.Pod) bool {
+		for _, s := range pod.Status.ContainerStatuses {
+			if s.Name == "main" && s.State.Waiting == nil {
```

Comment:
Could reuse `common.MainContainerName` instead?

Reply:
Iget you, but it is SOOOOOOO long variable name